examples/app_dag/app.py

```diff
@@ -80,12 +80,10 @@ def __init__(self, models_paths):
     def run(self):
         # Step 1 and 2: Download and process the data.
         self.data_collector.run()
-        self.data_collector.stop()  # Stop the data_collector to reduce cost
```

Comment:
what was wrong with this?

Reply:
As the work was stopped.

`self.data_collector.df_data` was None.

Reply:
I don't have enough context TBH

Reply:
There are other stop calls in the loop below

Reply:
But that's only for `ModelWork` no?

Reply:
Normally, they were blocking, so it should have worked.